static/js/components/DiskDeviceSelector.tsx

```diff
@@ -0,0 +1,94 @@
+import React, {
+  ChangeEvent,
+  Dispatch,
+  FC,
+  SetStateAction,
+  useState,
+} from "react";
+import { Input, Select } from "@canonical/react-components";
+import { useQuery } from "@tanstack/react-query";
+import { NotificationHelper } from "../types/notification";
+import { queryKeys } from "../util/queryKeys";
+import { LxdDiskDevice } from "../types/device";
+import { fetchStorages } from "../api/storages";
+
+interface Props {
+  notify: NotificationHelper;
+  setDiskDevice: Dispatch<SetStateAction<LxdDiskDevice | null>>;
+}
+
+const DiskDeviceSelector: FC<Props> = ({ notify, setDiskDevice }) => {
+  const [pool, setPool] = useState<string>("");
+  const [path, setPath] = useState<string>("");
+
+  const { data: storagePools = [], error } = useQuery({
+    queryKey: [queryKeys.storage],
+    queryFn: fetchStorages,
+  });
+
+  if (error) {
+    notify.failure("Could not load storage pools.", error);
+  }
+
+  const getStoragePoolOptions = () => {
+    const options = storagePools.map((storagePool) => {
+      return {
+        label: storagePool.name,
+        value: storagePool.name,
+        disabled: false,
+      };
+    });
+    options.unshift({
+      label:
+        storagePools.length === 0
+          ? "No storage pool available"
+          : "Select option",
+      value: "",
+      disabled: true,
+    });
+    return options;
+  };
+
+  const updatePool = (v: ChangeEvent<HTMLSelectElement>) => {
+    setPool(v.target.value);
+    setDiskDevice({
+      path: path,
+      pool: v.target.value,
+      type: "disk",
+    });
+  };
+
+  const updatePath = (v: ChangeEvent<HTMLInputElement>) => {
+    setPath(v.target.value);
+    setDiskDevice({
+      path: v.target.value,
+      pool: pool,
+      type: "disk",
+    });
+  };
+
+  return (
+    <>
+      <h4>Root disk device</h4>
+      <Select
+        name="pool"
+        label="Select a storage pool"
+        options={getStoragePoolOptions()}
+        onChange={updatePool}
+        value={pool}
+        stacked
+      />
+      <Input
```

Comment:
We might want to hardcode this to / and then we won't need an input for the path.

Reply:
Sure, just one question: should this still be shown to the user (e.g. as a disabled `Input`, or in any other way), so that they are aware of the `path` value that is being applied, or should it be completely transparent since they cannot modify it anyway? Which approach is better?

Reply:
I think we should simplify the form and remove the field.